### ISSUE TITLE
The keyboard navigation is not responsive after clicking a line from the overview 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-code-manager#914]
STR:
1. Load https://code.addons-dev.allizom.org/en-US/compare/616913/versions/1688316...1688317/?path=META-INF%2Fmanifest.mf#D5 and click on the overview on one of the lines from 1 to 4
2. Use `n` to navigate to the next change 

Expected result:
The line with the changes is highlighted. 

Actual result:
The line clicked in the overview stays highlighted at step 1. 

Notes:
Also reproducible with linter messages. 
Reproducible on FF67(Win10). 

Clicking a line from the overview then `n` 
Also reproducible if I click a line after all the changes from the page (using `n` should take me to the next file which contains changes).

![stays at line3](https://user-images.githubusercontent.com/33448286/60662982-92e75e00-9e66-11e9-9c17-0ada445a1b64.gif)

**Comment by bobsilverberg** [comment]
This STR is triggering [this error](https://github.com/mozilla/addons-code-manager/blob/2019.07.04/src/reducers/versions.tsx#L565-L569). 

The code assumes that `currentAnchor` must be an anchor for a diff, but if one selects a line via the Code Overview (or even by just clicking on a line itself), then whatever anchor is added to the URL is assigned to `currentAnchor`, and that might not be an anchor for a diff. That is what is triggering the error mentioned above, and that aborts the navigation operation.

So the assumption has to be removed, and we need to deal with a case where `currentAnchor` is not an anchor for a diff in the file. That's easy enough to do, but the question is: what do we do in that case?

Currently, if we have no anchor, then we just go to the first diff in the file. That's probably what we should do in this case. If we wanted to get really fancy, we could try to find the next anchor in the current file that we believe is after the currently selected anchor, by comparing just the numeric part of the anchor, but I'm not sure if we really need to do that.

To sum up, here are the two possibilities for behaviour that I am proposing, which would happen when a user has manually selected a line of code (either via the Code Overview or by clicking on a line) and then presses `n` or `p` to go to the next or previous diff:
1. Go to the first diff in the file.
2. Go to the next/previous diff in the file with respect to the currently selecting line.

1 is easy, 2 is more complex, but likely doable.

@kumar303 and @wagnerand What do you think?

**Comment by wagnerand** [comment]
2 seems the more logical choice to me. I don't think we should do 1, as this could be quite confusing for reviewers. It would potentially mean they'd jump back to something they have already seen, when what they wanted is to go forward.

So, if/when we can let's do 2, until then I'd say let's not do 1.

**Comment by kumar303** [comment]
I agree ^ but if there's a simpler patch we can land just to avoid throwing an error then it would be good to land that and follow-up with the more complete fix.

**Comment by bobsilverberg** [comment]
> I agree ^ but if there's a simpler patch we can land just to avoid throwing an error then it would be good to land that and follow-up with the more complete fix.

Option 1 above is very simple and will avoid throwing the error, so I guess it makes sense to implement that and then address option 2 in a follow-up. I'll get started on that.

**Comment by ioanarusiczki** [comment]
@bobsilverberg It seems to be working just fine on dev - F68(Win10)

The only thing is not alright would be that while using `n` or `p` the `cose.sig` and `mozilla rsa` files are shown, files which have no changes. 
Will the [2.](https://github.com/mozilla/addons-code-manager/issues/914#issuecomment-508554395) fix that or should I file a separate issue? 

![no changes shown](https://user-images.githubusercontent.com/33448286/60975357-89f9ff00-a334-11e9-852f-e5004f0816d1.gif)
